docs/devel/operator/UidGidResolver.md

```diff
@@ -0,0 +1,37 @@
+---
+title: UidGidResolver
+weight: 100
+description: >
+  Resolving uid and gid to username and groupname
+---
+
+# UidGidResolver
+
+The `UidGidResolver` resolves user ids and group ids to their corresponding names.
+
+This is done by reading `/etc/passwd` and `/etc/group` on the host.
+Therefore any `uid` inside a container might not properly match the username inside the container.
+Since the path is hardcoded usernames provided through `ldap`, `nss-systemd`, systemd units with `DynamicUser=yes`, ... will not be resolved correctly.
+
+## Usage
+
+### Classic gadgets
+
+1. Implement the UidResolverInterface for the `event struct` to resolve UIDs.
+   The return type is a map which maps the requested `uid` to string pointers, where the username should be written to.
+    ```go
+    type UidResolverInterface interface {
+	    ResolveUids() map[uint32]*string
+    }
+    ```
+2. Implement the GidResolverInterface for the `event struct` to resolve GIDs.
+   The return type is a map which maps the requested `gid` to string pointers, where the groupname should be written to.
+    ```go
+    type GidResolverInterface interface {
+	    ResolveGids() map[uint32]*string
+    }
+    ```
```

Comment:
The interface was changed, right? I'd suggest not to document this part as we're trying to remove these builtin gadgets soon.